components/authz-service/storage/v2/postgres/postgres.go

```diff
@@ -241,11 +256,32 @@ func (p *pg) UpdatePolicy(ctx context.Context, pol *v2.Policy) (*v2.Policy, erro
 	}
 
 	// Project filtering handled in here. We'll return a 404 right away if we can't find
-	// the policy via ID as filtered by projects.
-	if _, err = p.queryPolicy(ctx, pol.ID, tx); err != nil {
+	// the policy via ID as filtered by projects. Also selects for update if in v2.1 mode
```

Comment:
I was not familiar with "select... for update" so using that phrase here and in SQL below did not enlighten me. I think this change will remind me next time I look at it what "select for update" actually does. 
```suggestion
	// the policy via ID as filtered by projects. Also locks relevant rows if in v2.1 mode
```